nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestCommentManager.java

```diff
@@ -319,6 +356,186 @@ public void testCommentManagerType() {
+        addSubscription(doc, "CommentRemoved");
+        transactionalFeature.nextTransaction();
+
+        publishAndVerifyEventNotification(() -> {
+            commentManager.deleteComment(session, createdComment.getId());
+            return commentDocModel;
+
+        }, COMMENT_REMOVED, DOCUMENT_REMOVED, 0);
+    }
+
+    @Test
+    public void shouldGetCommentedDocModel() {
+        DocumentModel domain = session.createDocumentModel("/", "domain", "Domain");
+        session.createDocument(domain);
+        DocumentModel doc = session.createDocumentModel(domain.getPathAsString(), "test", "File");
+        doc = session.createDocument(doc);
+
+        Comment comment = new CommentImpl();
+        comment.setAuthor("author");
+        comment.setText("any Comment message");
+        comment.setParentId(doc.getId());
+
+        Comment createdComment = commentManager.createComment(session, comment);
+        DocumentModel commentDocModel = session.getDocument(new IdRef(createdComment.getId()));
+        assertEquals(getCommentedDocRef(session, commentDocModel, false), doc.getRef());
+
+        comment = new CommentImpl();
+        comment.setAuthor("author");
+        comment.setText("I am a reply");
+        comment.setParentId(commentDocModel.getId());
+        Comment replyComment = commentManager.createComment(session, comment);
+        assertEquals( //
+                getCommentedDocRef(session, session.getDocument(new IdRef(replyComment.getId())), true), //
+                commentDocModel.getRef());
+    }
+
+    protected void publishAndVerifyEventNotification(Supplier<DocumentModel> supplier, String commentEventType,
+            String documentEventType, int expectedMailsCount) {
+        try (CapturingEventListener listener = new CapturingEventListener(commentEventType, documentEventType)) {
+            DocumentModel documentModel = supplier.get();
+            transactionalFeature.nextTransaction();
+
+            assertTrue(listener.hasBeenFired(commentEventType));
+            assertTrue(listener.hasBeenFired(documentEventType));
+
+            Event event = checkAndGetCommentEvent(commentEventType, listener);
+
+            checkDocumentEventContext(documentModel, event);
+        }
+    }
+
+    /**
+     * Gets and ensures that the published comment event will be correctly handled.
+     */
+    protected Event checkAndGetCommentEvent(String commentEventType, CapturingEventListener listener) {
+        List<Event> handledEvents = listener.getCapturedEvents()
+                                            .stream()
+                                            .filter(e -> commentEventType.equals(e.getName()))
+                                            .collect(Collectors.toList());
+
+        assertEquals(1, handledEvents.size());
+        Event expectedEvent = handledEvents.get(0);
+        assertEquals(commentEventType, expectedEvent.getName());
+
+        return expectedEvent;
+    }
```

Comment:
Not needed, it is called once from the method above and I think it reduces the readability.